RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -243,11 +621,195 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
+              E_p3     = theta_E_pair[layer].second[ind_1+3];
+              theta_p3 = theta_E_pair[layer].first[ind_1+3];
+            } else {
+              E_p3     = 0;
+              theta_p3 = 0;
+            }
+ 
+            if (ind_1-4 >= 0) {
+              E_m4     = theta_E_pair[layer].second[ind_1-4];
+              theta_m4 = theta_E_pair[layer].first[ind_1-4];
+            } else {
+              E_m4     = 0;
+              theta_m4 = 0;
+            }
+            if (static_cast<size_t>(ind_1+4) < theta_E_pair[layer].second.size()) {
+              E_p4     = theta_E_pair[layer].second[ind_1+4];
+              theta_p4 = theta_E_pair[layer].first[ind_1+4];
+            } else {
+              E_p4     = 0;
+              theta_p4 = 0;
+            }
+
+            sum_E_3Bin = E_m1 + E_cell_Max[layer+startPositionToFill] + E_p1;
+            sum_E_5Bin = sum_E_3Bin + E_m2 + E_p2;
+            sum_E_7Bin = sum_E_5Bin + E_m3 + E_p3;
+            sum_E_9Bin = sum_E_7Bin + E_m4 + E_p4;
+
+            theta2_E_3Bin = theta_m1 * theta_m1 * E_m1 + E_cell_Max_theta[layer+startPositionToFill] * E_cell_Max_theta[layer+startPositionToFill] * E_cell_Max[layer+startPositionToFill] + theta_p1 * theta_p1 * E_p1;
+            theta_E_3Bin = theta_m1 * E_m1 + E_cell_Max_theta[layer+startPositionToFill] * E_cell_Max[layer+startPositionToFill] + theta_p1 * E_p1;
+            theta2_E_5Bin = theta2_E_3Bin + theta_m2 * theta_m2 * E_m2 + theta_p2 * theta_p2 * E_p2;
+            theta_E_5Bin = theta_E_3Bin + theta_m2 * E_m2 + theta_p2 * E_p2;
+            theta2_E_7Bin = theta2_E_5Bin + theta_m3 * theta_m3 * E_m3 + theta_p3 * theta_p3 * E_p3;
+            theta_E_7Bin = theta_E_5Bin + theta_m3 * E_m3 + theta_p3 * E_p3;
+            theta2_E_9Bin = theta2_E_7Bin + theta_m4 * theta_m4 * E_m4 + theta_p4 * theta_p4 * E_p4;
+            theta_E_9Bin = theta_E_7Bin + theta_m4 * E_m4 + theta_p4 * E_p4;
+
+            double _w_theta_3Bin = sqrt(fabs(theta2_E_3Bin / sum_E_3Bin - std::pow(theta_E_3Bin / sum_E_3Bin, 2)));
+            double _w_theta_5Bin = sqrt(fabs(theta2_E_5Bin / sum_E_5Bin - std::pow(theta_E_5Bin / sum_E_5Bin, 2)));
+            double _w_theta_7Bin = sqrt(fabs(theta2_E_7Bin / sum_E_7Bin - std::pow(theta_E_7Bin / sum_E_7Bin, 2)));
+            double _w_theta_9Bin = sqrt(fabs(theta2_E_9Bin / sum_E_9Bin - std::pow(theta_E_9Bin / sum_E_9Bin, 2)));
+
+            if (std::isnan(_w_theta_3Bin))    _w_theta_3Bin = 0.;
+            width_theta_3Bin[layer+startPositionToFill] = _w_theta_3Bin;
+            if (std::isnan(_w_theta_5Bin))    _w_theta_5Bin = 0.;
+            width_theta_5Bin[layer+startPositionToFill] = _w_theta_5Bin;
+            if (std::isnan(_w_theta_7Bin))    _w_theta_7Bin = 0.;
+            width_theta_7Bin[layer+startPositionToFill] = _w_theta_7Bin;
+            if (std::isnan(_w_theta_9Bin))    _w_theta_9Bin = 0.;
+            width_theta_9Bin[layer+startPositionToFill] = _w_theta_9Bin;
```

Comment:
does it ever happen that the energy sums in 3-5-7-9 bins are zero? This should not happen as long as you have a maximum energy cell with E>0... I'd rather check this or that the energy sums are zero, rather than correcting the nan shower shapes later (they could be nan due to some other bugs and we would not be able to figure out)